ui-discover/src/main/java/app/tivi/home/discover/Discover.kt

```diff
@@ -0,0 +1,424 @@
+import app.tivi.common.compose.AutoSizedCircularProgressIndicator
+import app.tivi.common.compose.Carousel
+import app.tivi.common.compose.IconResource
+import app.tivi.common.compose.LogCompositions
+import app.tivi.common.compose.PosterCard
+import app.tivi.common.compose.ProvideDisplayInsets
+import app.tivi.common.compose.TiviDateFormatterAmbient
+import app.tivi.common.compose.onSizeChanged
+import app.tivi.common.compose.rememberMutableState
+import app.tivi.common.compose.statusBarsPadding
+import app.tivi.data.entities.Episode
+import app.tivi.data.entities.Season
+import app.tivi.data.entities.TiviShow
+import app.tivi.data.entities.TmdbImageEntity
+import app.tivi.data.entities.TraktUser
+import app.tivi.data.resultentities.EntryWithShow
+import app.tivi.trakt.TraktAuthState
+import app.tivi.util.TiviDateFormatter
+import com.google.android.material.composethemeadapter.MdcTheme
+import dev.chrisbanes.accompanist.coil.CoilImage
+
+internal fun ViewGroup.composeDiscover(
+    state: LiveData<DiscoverViewState>,
+    actioner: (DiscoverAction) -> Unit,
+    tiviDateFormatter: TiviDateFormatter,
+    textCreator: DiscoverTextCreator
+): Any = setContent(Recomposer.current()) {
+    Providers(
+        TiviDateFormatterAmbient provides tiviDateFormatter,
+        DiscoverTextCreatorAmbient provides textCreator
+    ) {
+        MdcTheme {
+            LogCompositions("MdcTheme")
+
+            ProvideDisplayInsets {
+                LogCompositions("ProvideInsets")
+                val viewState by state.observeAsState()
+                if (viewState != null) {
+                    LogCompositions("ViewState observeAsState")
+                    Discover(viewState!!, actioner)
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalLazyDsl::class)
+@Composable
+fun Discover(
+    state: DiscoverViewState,
+    actioner: (DiscoverAction) -> Unit
+) {
+    Surface(Modifier.fillMaxSize()) {
+        Stack(Modifier.fillMaxSize()) {
+            var appBarHeight by rememberMutableState { 0 }
+
+            LazyColumn(Modifier.fillMaxSize()) {
+                item {
+                    val height = with(DensityAmbient.current) { appBarHeight.toDp() } + 16.dp
+                    Spacer(Modifier.preferredHeight(height))
+                }
+
+                state.nextEpisodeWithShowToWatched?.also { nextEpisodeToWatch ->
+                    item {
+                        Header(title = stringResource(R.string.discover_keep_watching_title))
+                    }
+                    item {
+                        NextEpisodeToWatch(
+                            show = nextEpisodeToWatch.show,
+                            poster = nextEpisodeToWatch.poster,
+                            season = nextEpisodeToWatch.season,
+                            episode = nextEpisodeToWatch.episode,
+                            modifier = Modifier.fillMaxWidth().clickable {
+                                actioner(
```

Comment:
This looks odd?! An onClick defining UI?

Reply:
Yeaaahhhhh, something I generally need to fix in Tivi. Basically I'm missing the 'processor' / 'actor' bit of MVI currently.